kernel/cls/src/lib.rs

```diff
@@ -7,3 +7,20 @@
 extern crate alloc;
 
 pub use cls_macros::cpu_local;
+
+pub trait Guard: sealed::Sealed {}
```

Comment:
Good solution! Just needs some docs.

Also, I'd vote to give it a more descriptive name, e.g., `AtomicityGuard` or something to do with ensuring atomicity (also mentioned in my other comment)